Make header retrieval case-insensitive

GMTA opened the pull request:
At the moment, case folding is applied to header names rendering them exclusively in lower-case. This change updates the Parser->getRawHeader($name) function and related functions to always fold the header name to lower-case.

This prevents confusion when someone tries to match the header name exactly as it's stored in an RFC(2)822 file and matches the functionality offered by a lot of other libraries that deal with headers.

NB: Adding a test for this would be trivial, however, I was not sure how to kick them off myself. That would be a nice future addition to README.md :-)

eXorus commented:
Hello,

It simple you just need to add one function in tests/ParserTest.php, something like that:

```
    public function testHeaderRetrievalIsCaseInsensitive()
    {
        //Init
        $file = __DIR__.'/mails/m0001';

        //Load From Path
        $Parser = new Parser();
        $Parser->setPath($file);

        $this->assertEquals($Parser->getRawHeader('From'), $Parser->getRawHeader('from'));
        $this->assertEquals($Parser->getHeader('From'), $Parser->getHeader('from'));
        $this->assertEquals($Parser->getAddresses('To'), $Parser->getAddresses('to'));

    }
```

This test should failed before your PR.

Could you fix "case-insensitve" to "case-insensitive" ?

Thanks

GMTA commented:
Wow, thanks for writing the test for me. I was actually asking *how* to run the tests but that became clear after running a composer install.

I've added your test and fixed the typos.

Probably unrelated: one of the tests fails for me @ PHP 5.6.24-0+deb8u1:
`1) PhpMimeMailParser\ExceptionTest::testWrongCharset
Failed asserting that exception of type "PHPUnit_Framework_Error" is thrown.`

eXorus commented:
Nice good job man !!

eXorus commented:
I added more information to show how to contribute
